pandas/tests/frame/indexing/test_getitem.py

```diff
@@ -172,6 +174,66 @@ def test_getitem_iloc_two_dimensional_generator(self):
         expected = Series([5, 6], name="b", index=[1, 2])
         tm.assert_series_equal(result, expected)
 
+    def test_getitem_iloc_dateoffset_days(self):
+        # GH 46671
+        df = DataFrame(
+            list(range(10)),
+            index=date_range("01-01-2022", periods=10, freq=DateOffset(days=1)),
+        )
+        result = df.loc["2022-01-01":"2022-01-03"]
+        expected = DataFrame(
+            [0, 1, 2],
+            index=DatetimeIndex(
+                ["2022-01-01", "2022-01-02", "2022-01-03"],
+                dtype="datetime64[ns]",
+                freq=DateOffset(days=1),
+            ),
+        )
+        tm.assert_frame_equal(result, expected)
+
+        df = DataFrame(
+            list(range(10)),
+            index=date_range(
+                "01-01-2022", periods=10, freq=DateOffset(days=1, hours=2)
+            ),
+        )
+        result = df.loc["2022-01-01":"2022-01-03"]
+        expected = DataFrame(
+            [0, 1, 2],
+            index=DatetimeIndex(
+                ["2022-01-01 00:00:00", "2022-01-02 02:00:00", "2022-01-03 04:00:00"],
+                dtype="datetime64[ns]",
+                freq=DateOffset(days=1, hours=2),
+            ),
+        )
+        tm.assert_frame_equal(result, expected)
+
+        df = DataFrame(
+            list(range(10)),
+            index=date_range("01-01-2022", periods=10, freq=DateOffset(minutes=3)),
+        )
+        result = df.loc["2022-01-01":"2022-01-03"]
+        expected = DataFrame(
+            [0, 1, 2, 3, 4, 5, 6, 7, 8, 9],
+            index=DatetimeIndex(
+                [
+                    "2022-01-01 00:00:00",
+                    "2022-01-01 00:03:00",
+                    "2022-01-01 00:06:00",
+                    "2022-01-01 00:09:00",
+                    "2022-01-01 00:12:00",
+                    "2022-01-01 00:15:00",
+                    "2022-01-01 00:18:00",
+                    "2022-01-01 00:21:00",
+                    "2022-01-01 00:24:00",
+                    "2022-01-01 00:27:00",
+                ],
+                dtype="datetime64[ns]",
+                freq=DateOffset(minutes=3),
+            ),
+        )
```

Comment:
This expected is just the original `df` ? (so can we reuse that instead?)

Reply:
done